tools/src/main/java/org/apache/kafka/tools/ConsumerPerformance.java

```diff
@@ -0,0 +1,394 @@
+                    "See java.text.SimpleDateFormat for options.")
+                .withRequiredArg()
+                .describedAs("date format")
+                .ofType(String.class)
+                .defaultsTo("yyyy-MM-dd HH:mm:ss:SSS");
+            hideHeaderOpt = parser.accepts("hide-header", "If set, skips printing the header for the stats");
+            try {
+                options = parser.parse(args);
+            } catch (OptionException e) {
+                CommandLineUtils.printUsageAndExit(parser, e.getMessage());
+                return;
+            }
+            if (options != null) {
+                if (options.has(numThreadsOpt) || options.has(numFetchersOpt))
+                    System.out.println("WARNING: option [threads] and [num-fetch-threads] have been deprecated and will be ignored by the test");
+                CommandLineUtils.maybePrintHelpOrVersion(this, "This tool is used to verify the consumer performance.");
+                CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, numMessagesOpt);
+            }
+        }
+
+        public boolean printMetrics() {
+            return options.has(printMetricsOpt);
+        }
+
+        public String brokerHostsAndPorts() {
+            return options.valueOf(options.has(bootstrapServerOpt) ? bootstrapServerOpt : brokerListOpt);
+        }
+
+        public Properties props() throws IOException {
+            Properties props = (options.has(consumerConfigOpt))
+                ? Utils.loadProps(options.valueOf(consumerConfigOpt))
+                : new Properties();
+            props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerHostsAndPorts());
+            props.put(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(groupIdOpt));
+            props.put(ConsumerConfig.RECEIVE_BUFFER_CONFIG, options.valueOf(socketBufferSizeOpt).toString());
+            props.put(ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG, options.valueOf(fetchSizeOpt).toString());
+            props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+                options.has(resetBeginningOffsetOpt) ? "latest" : "earliest");
+            props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+            props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+            props.put(ConsumerConfig.CHECK_CRCS_CONFIG, "false");
+            if (props.getProperty(ConsumerConfig.CLIENT_ID_CONFIG) == null)
+                props.put(ConsumerConfig.CLIENT_ID_CONFIG, "perf-consumer-client");
+            return props;
+        }
+
+        public Set<String> topic() {
+            return Collections.singleton(options.valueOf(topicOpt));
+        }
+
+        public long numMessages() {
+            return options.valueOf(numMessagesOpt);
+        }
+
+        public int reportingIntervalMs() {
+            int value = options.valueOf(reportingIntervalOpt);
+            if (value <= 0)
```

Comment:
Should this be checked at initialization time?

Reply:
Looks like it's not.

Reply:
I think the scala version performs the validation at initialisation, before starting the test. That is a minor non-impactful difference though.

Reply:
You are absolutely right: same code, different behavior. Thanks for spotting this. With my last commit I added validation to that option and default value in the description.